app/components/osc-webhook-triggers/osc-webhook-triggers.html

```diff
@@ -1,38 +1,53 @@
-<ng-form name="secretsForm" class="add-webhook">
+<ng-form name="$ctrl.secretsForm" class="add-webhook">
   <div ng-repeat="trigger in $ctrl.webhookTriggers">
     <div class="add-webhook-row">
-      <ui-select 
-        ng-model="trigger.data.type"
-        ng-disabled="$ctrl.isDeprecated(trigger)"
-        on-select="$ctrl.triggerTypeChange(trigger)"
-        search-enabled="false"
-        title="Select a webhook type"
-        class="select-webhook-type"
-        flex>
-        <ui-select-match placeholder="Webhook type">
-          {{ $select.selected.label }}
-        </ui-select-match>
-        <ui-select-choices repeat="option.label as option in $ctrl.webhookTypesOptions">
-          {{ option.label }}
-        </ui-select-choices>
-      </ui-select>
+      <div class="select-webhook-type">
+        <ui-select 
+          ng-model="trigger.data.type"
+          name="triggerType{{$index}}"
+          ng-disabled="$ctrl.isDeprecated(trigger)"
+          on-select="$ctrl.triggerTypeChange(trigger)"
+          search-enabled="false"
+          title="Select a webhook type"
+          ng-class="{'has-error': $ctrl.missingInputError('triggerType', $index, trigger.data.type, trigger.data[trigger.data.type.toLowerCase()].secretReference.name) }"
+          focus-on="triggerTypeFocus{{$index}}"
```

Comment:
It's unclear to me where the `focus-on` directive is defined. Is this a dependency?

Reply:
Thats a ui-select directive

Reply:
OK, I see. Thanks!